### PR TITLE
Documentation : Jackson Java 8 modules and a small correction

### PR DESCRIPTION
**Pull request by loicmathieu**
Fix a small documentation error.

Add a reference to the Java 8 modules that Quarkus automatically add in the rest json guide.

### CODE REVIEW

File: docs/src/main/asciidoc/rest-json-guide.adoc [hunk 1]
```diff
@@ -63,6 +63,8 @@ mvn io.quarkus:quarkus-maven-plugin:{quarkus-version}:create \
     -Dextensions="resteasy-jackson"
 ----
 
+To improve user experience, Quarkus registers the three Jackson https://github.com/FasterXML/jackson-modules-java8[Java 8 modules] so you don't need to do it manually.
```
[the remark this answers is not in the document]
Comment (loicmathieu) [reply docs/src/main/asciidoc/rest-json-guide.adoc]: Yeah, first time I wrote a document sentence without any correction needed !
I must have improve my english :)

[the remark this answers is not in the document]
Comment (loicmathieu) [reply docs/src/main/asciidoc/rest-json-guide.adoc]: It's a work in progress ...

File: docs/src/main/asciidoc/cdi-reference.adoc [hunk 2]
```diff
@@ -484,7 +484,7 @@ In order to see more information about which beans are being removed,
 you can enable additional logging via the following line in your `application.properties`.
 
 ----
-quarkus.log.category."io.quarkus.arc.producer".level=DEBUG
+quarkus.log.category."io.quarkus.arc.processor".level=DEBUG
```
[the remark this answers is not in the document]
Comment (mkouba) [reply docs/src/main/asciidoc/cdi-reference.adoc]: No, this is not correct. For `application.properties` file can only be used for run time configuration of logging and beans are removed during build. So the only way to show the debug messages is to configure the build tool used, i.e. Maven through `mvn -X` and grep "Removed unused".

**Review by mkouba — DISMISSED** [review]
Pls fix the logging info in `cdi-reference.adoc`.

File: docs/src/main/asciidoc/cdi-reference.adoc (hunk 2 above)
[the remark this answers is not in the document]
Comment (loicmathieu) [reply docs/src/main/asciidoc/cdi-reference.adoc]: @mkouba it works in dev mode (I used it) as the build is done at runtime.
I propose to change it this way:
```
When using the dev mode (running `mvn clean compile quarkus:dev`), you can see more information about which beans are being removed 
by enabling additional logging via the following line in your `application.properties`.
```

**Comment by gsmet** [comment]
@loicmathieu can you squash the last commit, please? Let's try to have semantic commits. Thanks!

**Comment by loicmathieu** [comment]
@gsmet done

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
Merged, thanks!

**Pull request merged**